Branch-SDK/src/io/branch/referral/util/CommerceEvent.java

```diff
@@ -0,0 +1,82 @@
+package io.branch.referral.util;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by Evan Groth on 12/21/16.
+ */
+public class CommerceEvent {
+    public BigDecimal revenue;
+    public String currency;
+    public String transactionID;
+    public BigDecimal shipping;
+    public BigDecimal tax;
+    public String coupon;
+    public String affiliation;
+    public List<Product> products;
+
+    public CommerceEvent() {
+        this.revenue = new BigDecimal(0);
+        this.currency = "";
+        this.transactionID = "";
+        this.shipping = new BigDecimal(0);
+        this.tax = new BigDecimal(0);
+        this.coupon = "";
+        this.affiliation = "";
+        this.products = new ArrayList<>();
+    }
+
+    public CommerceEvent(BigDecimal revenue, String currency, String transactionID, BigDecimal shipping, BigDecimal tax, String coupon, String affiliation, List<Product> products) {
+        this.revenue = revenue;
+        this.currency = currency;
+        this.transactionID = transactionID;
+        this.shipping = shipping;
+        this.tax = tax;
+        this.coupon = coupon;
+        this.affiliation = affiliation;
+        this.products = products;
+    }
+
+    public CommerceEvent(BigDecimal revenue, String currency, String transactionID, BigDecimal shipping, BigDecimal tax, String coupon, String affiliation, Product product) {
```

Comment:
We have currenctType enumeration in SDK which is already used in BUO. The "string" for currency name can be replaced with it